Polygon_mesh_processing/include/CGAL/Envelope.h

```diff
@@ -0,0 +1,1905 @@
+    *                     must be available in `PolygonMesh`.}
+    *   \cgalParamNEnd
+    *
+    *   \cgalParamNBegin{vertex_index_map}
+    *     \cgalParamDescription{a property map associating to each vertex of `tmesh` a unique index between `0` and `num_vertices(tmesh) - 1`}
+    *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+    *                    as key type and `std::size_t` as value type}
+    *     \cgalParamDefault{an automatically indexed internal map}
+    *   \cgalParamNEnd
+    *
+    * \cgalNamedParamsEnd
+    *
+    * \note The triangle mesh gets copied internally, that is it can be modifed after having passed as argument,
+    *       while the queries are performed
+    */
+
+  template <typename TriangleMesh, typename NamedParameters>
+  Envelope(const TriangleMesh& tmesh,
+           double epsilon,
+           const NamedParameters& np)
+  {
+    using parameters::choose_parameter;
+    using parameters::get_parameter;
+
+
+    typename GetVertexPointMap<TriangleMesh, NamedParameters>::const_type
+      vpm = choose_parameter(get_parameter(np, internal_np::vertex_point),
+                             get_const_property_map(CGAL::vertex_point, tmesh));
+
+    typedef typename GetInitializedVertexIndexMap<TriangleMesh, NamedParameters>::const_type VertexIndexMap;
+    VertexIndexMap vim = CGAL::get_initialized_vertex_index_map(tmesh, np);
+
+    env_vertices.reserve(num_vertices(tmesh));
+    env_faces.reserve(num_faces(tmesh));
+
+    for(typename boost::graph_traits<TriangleMesh>::vertex_descriptor v : vertices(tmesh)){
+      env_vertices.emplace_back(get(vpm, v));
+    }
+
+    for(typename boost::graph_traits<TriangleMesh>::face_descriptor f : faces(tmesh)){
+      typename boost::graph_traits<TriangleMesh>::halfedge_descriptor h = halfedge(f, tmesh);
+      int i = get(vim, source(h, tmesh));
+      int j = get(vim, target(h, tmesh));
+      int k = get(vim, target(next(h, tmesh), tmesh));
+
+      Vector3i face = { i, j, k };
+      env_faces.push_back(face);
+    }
+    init(epsilon);
+  }
+
+
+  template <typename TriangleMesh>
+  Envelope(const TriangleMesh& tmesh,
+           double epsilon)
+    : Envelope(tmesh, epsilon, parameters::all_default())
+  {}
+
+
+private:
+
+  void init(double epsilon)
+  {
+    halfspace_generation(env_vertices, env_faces, halfspace, bounding_boxes, epsilon);
+
+    Datum_map<GeomTraits> datum_map(bounding_boxes);
+    Point_map<GeomTraits> point_map(bounding_boxes);
+
+    // constructs AABB tree
+    tree.insert(boost::counting_iterator<std::size_t>(0),
+                boost::counting_iterator<std::size_t>(bounding_boxes.size()),
+                datum_map,
+                point_map);
+    tree.build();
+  }
+
+
+  struct INDEX
+  {
+    int Pi;
+    std::vector<int> FACES;
+  };
+
+
+  // was marked todo???
```

Comment:
garbage?

Reply:
This was a comment in the original code.